---
title: "Combined use of eDNA metabarcoding and visual surveys to monitor rocky intertidal biodiversity"
format: html
embed-resources: true
editor: visual
author: Simons D-L, Hipperson H, Webb TJ, Spencer M, Mieszkowska N
chunk_output_type: inline
toc: true
message: false
warning: false
---

## Introduction

This pipeline replicates all analyses in Simons *et al* (2025) "Combined use of eDNA metabarcoding and visual surveys to monitor rocky intertidal biodiversity".

## Load packages

```{r}
#| label: load-packages
#| echo: true
#| results: false

packages <- c("devtools",
              "microbiome",
              "tidyverse",
              "vegan",
              "phyloseq",
              "eulerr",
              "cowplot",
              "ggVennDiagram",
              'lme4',
              "car",
              "emmeans",
              "ggpattern",
              "ggrepel",
              "sf")

new.packages <- packages[!(packages %in% installed.packages()[,"Package"])]
if(length(new.packages)) install.packages(new.packages)

invisible(lapply(packages, library, character.only = TRUE))
```

## Importing data and tidying

Let's load our phyloseq object and long data, which have already been tidied, decontaminated, filtered for target taxa, and checked for quality control through [this pipeline](https://github.com/dinaleighsimons/intertidal-eDNA-formatting-and-quality-control).

```{r}
#| label: load-data

visual <- read.csv("Input_Data/rocky_visual_data.csv")
eDNA <- read.csv("Input_Data/rocky_eDNA_long.csv", row.names = 1)

load("Input_Data/phylo_MarClim.RData")
load("Input_Data/phylo_rocky_eDNA.RData")
```

Let's first explore our two `phyloseq` objects.

```{r}
#| label: explore-data

phylo_MarClim
phylo_rocky_eDNA
```

The eDNA data contains sites which visual surveys weren't conducted at. Let's remove them.

```{r}
#| label: remove-sites-no-visual

# define sites with no visual survey
sites_no_visual <- c("Borwick, Yesnaby", "Martin's Haven", "Skomer Island")

# subset the long
eDNA <- subset(eDNA, !verbatimLocality %in% sites_no_visual)

# subset the phylo object
phylo_rocky_eDNA <- subset_samples(phylo_rocky_eDNA,
                                        !verbatimLocality %in% sites_no_visual)
phylo_rocky_eDNA
```

We need to fix some names so that they match in both data sets.

```{r}
#| label: fix-names-long

# fix unmatched names in long eDNA
eDNA$valid_name[eDNA$valid_name == "Mytilus"] <- "Mytilus spp"
eDNA$valid_name[eDNA$valid_name == "Mytilus edulis"] <- "Mytilus spp"
eDNA$valid_name[eDNA$valid_name == "Codium"] <- "Codium spp"
eDNA$valid_name[eDNA$valid_name == "Steromphala umbilicaris"] <- "Steromphala umbilicalis"

# fix unmatched names in long visual
visual$valid_name[visual$valid_name == "Mytilus"] <- "Mytilus spp"
visual$valid_name[visual$valid_name == "Codium"] <- "Codium spp"
```

### Combine (indirect, long)

We are going to combined the eDNA and visual data into the same data set. Let's start with the long format first. We will do this at three different levels (species, genus and family) by performing direct and indirect matches. Indirect matches first.

First, let's match the capitalization of the names.

```{r}
#| label: tidying-capitalisation

#correct capitalisation in visual
visual$sciname <- sapply(strsplit(visual$sciname, " "),
                         #split taxa into two
                         function(x)
                           paste(toupper(substr(x[1], 1, 1)) %>% #converts genus to upper
                                   paste0(tolower(
                                     substr(x[1], 2, nchar(x[1]))
                                   )), tolower(paste(x[-1], collapse = " "))))
```

Let's filter the eDNA data using the visual survey species list. For the indirect matches, we are going to keep all the taxa which match any of the taxa listed in the MarClim list at all levels. This means we'll get a broader range of species within the eDNA data set compared to visual at higher taxonomic levels.

```{r}
#| label: filter-eDNA-data-long-indirect

# make names factors
visual$valid_name <- as.factor(visual$valid_name)
eDNA$valid_name <- as.factor(eDNA$valid_name)

# long - match to species, genus, family
visual_species <- unique(visual$valid_name)
eDNA_species <- subset(eDNA, valid_name %in% visual_species)
dim(eDNA_species)

visual_genera <- unique(visual$genus)
eDNA_genus <- subset(eDNA, genus %in% visual_genera)
dim(eDNA_genus)

visual_family <- unique(visual$family)
eDNA_family <- subset(eDNA, family %in% visual_family)
dim(eDNA_family)

visual_phylum <- unique(visual$phylum)
eDNA_phylum <- subset(eDNA, phylum %in% visual_phylum)
dim(eDNA_phylum)
```

Species-level combining.

First, let's format the visual data so we combine it later on.

```{r}
#| label: visual-format-species-long-indirect

## select relevant columns
visual_subset <- visual %>% dplyr::select(c(
  valid_name,
  localityID,
  Count,
  country,
  genus,
  verbatimLocality,
  phylum
))

## add method column
visual_subset$method <- "Visual"

## rename columns
colnames(visual_subset) <- c("taxa", "localityID", "count", "region", "genus", "site", "phylum", "method")

## add columns which are needed later for the eDNA joining
visual_subset <- visual_subset %>%
  mutate(pa = ifelse(count > 0, 1, 0),
         num_samples = NA,
         num_samples_found = NA,
         prop_samples_found = NA)
```

```{r}
#| label: eDNA-format-indirect-species

# Calculate number of samples which detected each species per site

## create pa variable
eDNA_species <- eDNA_species %>%
  mutate(pa = ifelse(reads > 0, 1, 0))

## Convert multiple columns to factor
eDNA_species <- eDNA_species %>%
  mutate(
    localityID = factor(localityID),
    fieldID = factor(fieldID),
    sampleType = factor(sampleType),
    taxa = factor(valid_name)
  )

## summarise to calculate proportions of detection at a site level
eDNA_site_level <- eDNA_species %>%
    dplyr::group_by(localityID, valid_name) %>%  # Group by site and taxa
    dplyr::summarise(
        num_samples = n_distinct(fieldID),
        num_samples_found = sum(pa > 0, na.rm = TRUE),
        prop_samples_found = sum(pa > 0, na.rm = TRUE)/n_distinct(fieldID)
    ) %>%
    ungroup()

## rename columns
colnames(eDNA_site_level) <- c("localityID", "taxa", "num_samples", "num_samples_found", "prop_samples_found")

## get columns of interest
eDNA_subset <- eDNA_species %>% dplyr::select(c(
  valid_name,
  localityID,
  reads,
  country,
  genus,
  verbatimLocality,
  pa,
  phylum
))

## rename columns
colnames(eDNA_subset) <- c("taxa",
                           "localityID",
                           "count",
                           "region",
                           "genus",
                           "site",
                           "pa",
                           "phylum")

## sum number of reads
eDNA_subset_site <- eDNA_subset %>% 
  dplyr::group_by(localityID, taxa) %>% # can add primer here if interested
  dplyr::summarise(
    count = sum(count))

## add method
eDNA_subset_site$method <- "eDNA"

## join proportions and summed counts together 
eDNA_join <- left_join(eDNA_subset_site,
                       eDNA_site_level,
                       by = c("taxa", "localityID"))

## get meta data back and add in
eDNA_subset_meta <- eDNA_subset %>% dplyr::select(c(localityID, region, genus, taxa, site, phylum)) %>%
  unique()

eDNA_join_meta <- left_join(eDNA_join,
                       eDNA_subset_meta,
                       by = c("taxa", "localityID"))

## recalculate PA
eDNA_join_meta <- eDNA_join_meta %>%
  mutate(pa = ifelse(count > 0, 1, 0))

## select relevant columns only for the final data
eDNA_join_final <- eDNA_join_meta %>%  select(c(taxa, localityID, count, region, genus, site, phylum, method, pa, num_samples, num_samples_found, prop_samples_found))

## check pa and counts align
any(eDNA_join_final$num_samples_found > 0 & eDNA_join_final$count == 0) #should be false
```

```{r}
#| label: combine-methoods-species-long-indirect

# check columns align - should all be true
colnames(visual_subset) == colnames(eDNA_join_final)

# combine
combined_data<- rbind(visual_subset, eDNA_join_final)

# remove any white space
combined_data$taxa <- trimws(combined_data$taxa)

# remove any duplicates
combined_data <- unique(combined_data)

# fix rogue name
combined_data$taxa <- gsub("Halichondria \\(Halichondria\\) panicea", "Halichondria panicea", combined_data$taxa)

# save
write.csv(combined_data, "Processed_data/combined_methods_long_data.csv")
```

Genus-level combining.

First, let's format the visual data to genus level so we combine it later on.

```{r}
#| label: visual-format-genus-long

# select relevant columns
visual_subset_genus <- visual %>% dplyr::select(c(
  valid_name,
  localityID,
  Count,
  country,
  genus,
  verbatimLocality,
  phylum
))

# add method column
visual_subset_genus$method <- "Visual"

# rename columns
colnames(visual_subset_genus) <- c("taxa", "localityID", "count", "region", "genus", "site", "phylum", "method")

# Compute unique present taxa per genus + site
num_taxa_summary <- visual_subset_genus %>%
  filter(count > 0) %>%
  group_by(genus, site) %>%
  summarise(num_taxa_unique = n_distinct(taxa), .groups = "drop")

# add observations from different species but same genus together, cal richness
visual_subset_genus <- visual_subset_genus %>%
  group_by(genus, site) %>%
  summarise(
    localityID = first(localityID),
    count = sum(count),
    region = first(region),
    phylum = first(phylum),
    method = first(method),
    .groups = 'drop'
  )

# Join back to original data (keeps absences)
visual_subset_genus <- visual_subset_genus %>%
  left_join(num_taxa_summary, by = c("genus", "site")) %>% 
  mutate(num_taxa_unique = replace_na(num_taxa_unique, 0))

# correct the order of columns
visual_subset_genus <- visual_subset_genus %>%
  select(localityID, count, region, genus, site, phylum, method,num_taxa_unique)

# add columns which are needed later for the eDNA joining
visual_subset_genus <- visual_subset_genus %>%
  mutate(pa = ifelse(count > 0, 1, 0),
         num_samples = NA,
         num_samples_found = NA,
         prop_samples_found = NA)
```

```{r}
#| label: eDNA-format-genus-long-indirect

# Calculate number of samples which detected each species per site

## create pa variable
eDNA_genus <- eDNA_genus %>%
  mutate(pa = ifelse(reads > 0, 1, 0))

## Convert multiple columns to factor
eDNA_genus <- eDNA_genus %>%
  mutate(
    taxa = factor(taxa),
    localityID = factor(localityID),
    fieldID = factor(fieldID),
    sampleType = factor(sampleType),
    genus = factor(genus)
  )

## summarise to calculate proportions of detection at a site level
eDNA_site_level_genus <- eDNA_genus %>%
    dplyr::group_by(localityID, genus) %>%  # Group by site and taxa
    dplyr::summarise(
        num_samples = n_distinct(fieldID),
        num_samples_found = sum(pa > 0, na.rm = TRUE),
        prop_samples_found = sum(pa > 0, na.rm = TRUE)/n_distinct(fieldID)
    ) %>%
    ungroup()

## rename columns
colnames(eDNA_site_level_genus) <- c("localityID", "genus", "num_samples", "num_samples_found", "prop_samples_found")

## get columns of interest
eDNA_subset_genus <- eDNA_genus %>% dplyr::select(c(
    valid_name,
    localityID,
    reads,
    country,
    genus,
    verbatimLocality,
    pa,
    phylum
  ))

## rename columns
colnames(eDNA_subset_genus) <- c("taxa",
                                 "localityID",
                                 "count",
                                 "region",
                                 "genus",
                                 "site",
                                 "pa",
                                 "phylum")

## add method
eDNA_subset_genus$method <- "eDNA"

# Compute unique present taxa per genus + site
num_taxa_summary_genus <- eDNA_subset_genus %>%
  filter(count > 0) %>%
  group_by(genus, site) %>%
  summarise(num_taxa_unique = n_distinct(taxa), .groups = "drop")

# add observations from different species but same genus together, cal richness
eDNA_subset_genus <- eDNA_subset_genus %>%
  group_by(genus, site) %>%
  summarise(
    localityID = first(localityID),
    count = sum(count),
    region = first(region),
    phylum = first(phylum),
    method = first(method),
    .groups = 'drop'
  )

# Join back to original data (keeps absences)
eDNA_subset_genus <- eDNA_subset_genus %>%
  left_join(num_taxa_summary_genus, by = c("genus", "site")) %>% 
  mutate(num_taxa_unique = replace_na(num_taxa_unique, 0))

## join proportions and summed counts together 
eDNA_join_genus <- left_join(eDNA_subset_genus,
                              eDNA_site_level_genus,
                              by = c("genus", "localityID")) %>%
  mutate(pa = ifelse(count > 0, 1, 0))

## select relevant columns only for the final data
eDNA_join_final_genus <- eDNA_join_genus %>%  select(c(localityID, count, region, genus, site, phylum, method, num_taxa_unique, pa, num_samples, num_samples_found, prop_samples_found))

## check pa and counts align
any(eDNA_join_final_genus$num_samples_found > 0 & eDNA_join_final_genus$count == 0) #should be false
```

```{r}
#| label: combine-methods-genus-long-indirect

# check columns align - should all be true
colnames(visual_subset_genus) == colnames(eDNA_join_final_genus)

# combine
combined_data_genus<- rbind(visual_subset_genus, eDNA_join_final_genus)

# remove any white space
combined_data_genus$genus <- trimws(combined_data_genus$genus)

# remove any duplicates
combined_data_genus <- unique(combined_data_genus)
```

Family-level combining.

First, let's format the visual data to family level so we combine it later on.

```{r}
#| label: visual-format-family-long

# select relevant columns
visual_subset_family <- visual %>% dplyr::select(c(
  valid_name,
  localityID,
  Count,
  country,
  family,
  verbatimLocality,
  phylum
))

# add method column
visual_subset_family$method <- "Visual"

# rename columns
colnames(visual_subset_family) <- c("taxa", "localityID", "count", "region", "family", "site", "phylum", "method")

# Compute unique present taxa per family + site
num_taxa_summary <- visual_subset_family %>%
  filter(count > 0) %>%
  group_by(family, site) %>%
  summarise(num_taxa_unique = n_distinct(taxa), .groups = "drop")

# add observations from different species but same family together, cal richness
visual_subset_family <- visual_subset_family %>%
  group_by(family, site) %>%
  summarise(
    localityID = first(localityID),
    count = sum(count),
    region = first(region),
    phylum = first(phylum),
    method = first(method),
    .groups = 'drop'
  )

# Join back to original data (keeps absences)
visual_subset_family <- visual_subset_family %>%
  left_join(num_taxa_summary, by = c("family", "site")) %>% 
  mutate(num_taxa_unique = replace_na(num_taxa_unique, 0))

# correct the order of columns
visual_subset_family <- visual_subset_family %>%
  select(localityID, count, region, family, site, phylum, method,num_taxa_unique)

# add columns which are needed later for the eDNA joining
visual_subset_family <- visual_subset_family %>%
  mutate(pa = ifelse(count > 0, 1, 0),
         num_samples = NA,
         num_samples_found = NA,
         prop_samples_found = NA)
```

```{r}
#| label: eDNA-format-family-long-indirect

# Calculate number of samples which detected each species per site

## create pa variable
eDNA_family <- eDNA_family %>%
  mutate(pa = ifelse(reads > 0, 1, 0))

## Convert multiple columns to factor
eDNA_family <- eDNA_family %>%
  mutate(
    taxa = factor(taxa),
    localityID = factor(localityID),
    fieldID = factor(fieldID),
    sampleType = factor(sampleType),
    family = factor(family)
  )

## summarise to calculate proportions of detection at a site level
eDNA_site_level_family <- eDNA_family %>%
    dplyr::group_by(localityID, family) %>%  # Group by site and taxa
    dplyr::summarise(
        num_samples = n_distinct(fieldID),
        num_samples_found = sum(pa > 0, na.rm = TRUE),
        prop_samples_found = sum(pa > 0, na.rm = TRUE)/n_distinct(fieldID)
    ) %>%
    ungroup()

## rename columns
colnames(eDNA_site_level_family) <- c("localityID", "family", "num_samples", "num_samples_found", "prop_samples_found")

## get columns of interest
eDNA_subset_family <- eDNA_family %>% dplyr::select(c(
    valid_name,
    localityID,
    reads,
    country,
    family,
    verbatimLocality,
    pa,
    phylum
  ))

## rename columns
colnames(eDNA_subset_family) <- c("taxa",
                                 "localityID",
                                 "count",
                                 "region",
                                 "family",
                                 "site",
                                 "pa",
                                 "phylum")

## add method
eDNA_subset_family$method <- "eDNA"

# Compute unique present taxa per family + site
num_taxa_summary_family <- eDNA_subset_family %>%
  filter(count > 0) %>%
  group_by(family, site) %>%
  summarise(num_taxa_unique = n_distinct(taxa), .groups = "drop")

# add observations from different species but same family together, cal richness
eDNA_subset_family <- eDNA_subset_family %>%
  group_by(family, site) %>%
  summarise(
    localityID = first(localityID),
    count = sum(count),
    region = first(region),
    phylum = first(phylum),
    method = first(method),
    .groups = 'drop'
  )

# Join back to original data (keeps absences)
eDNA_subset_family <- eDNA_subset_family %>%
  left_join(num_taxa_summary_family, by = c("family", "site")) %>% 
  mutate(num_taxa_unique = replace_na(num_taxa_unique, 0))

## join proportions and summed counts together 
eDNA_join_family <- left_join(eDNA_subset_family,
                              eDNA_site_level_family,
                              by = c("family", "localityID")) %>%
  mutate(pa = ifelse(count > 0, 1, 0))

## select relevant columns only for the final data
eDNA_join_final_family <- eDNA_join_family %>%  select(c(localityID, count, region, family, site, phylum, method, num_taxa_unique, pa, num_samples, num_samples_found, prop_samples_found))

## check pa and counts align
any(eDNA_join_final_family$num_samples_found > 0 & eDNA_join_final_family$count == 0) #should be false
```

```{r}

#| label: combine-methods-family-long-indirect

# check columns align - should all be true
colnames(visual_subset_family) == colnames(eDNA_join_final_family)

# combine
combined_data_family<- rbind(visual_subset_family, eDNA_join_final_family)

# remove any white space
combined_data_family$family <- trimws(combined_data_family$family)

# remove any duplicates
combined_data_family <- unique(combined_data_family)
```

Phylum level

```{r}

#| label: visual-format-phylum-long

# select relevant columns
visual_subset_phylum <- visual %>% dplyr::select(c(
  valid_name,
  localityID,
  Count,
  country,
  phylum,
  verbatimLocality,
))

# add method column
visual_subset_phylum$method <- "Visual"

# rename columns
colnames(visual_subset_phylum) <- c("taxa", "localityID", "count", "region", "phylum", "site", "method")

# Compute unique present taxa per phylum + site
num_taxa_summary <- visual_subset_phylum %>%
  filter(count > 0) %>%
  group_by(phylum, site) %>%
  summarise(num_taxa_unique = n_distinct(taxa), .groups = "drop")

# add observations from different species but same phylum together, cal richness
visual_subset_phylum <- visual_subset_phylum %>%
  group_by(phylum, site) %>%
  summarise(
    localityID = first(localityID),
    count = sum(count),
    region = first(region),
    method = first(method),
    .groups = 'drop'
  )

# Join back to original data (keeps absences)
visual_subset_phylum <- visual_subset_phylum %>%
  left_join(num_taxa_summary, by = c("phylum", "site")) %>% 
  mutate(num_taxa_unique = replace_na(num_taxa_unique, 0))

# correct the order of columns
visual_subset_phylum <- visual_subset_phylum %>%
  select(localityID, count, region, phylum, site, method,num_taxa_unique)

# add columns which are needed later for the eDNA joining
visual_subset_phylum <- visual_subset_phylum %>%
  mutate(pa = ifelse(count > 0, 1, 0),
         num_samples = NA,
         num_samples_found = NA,
         prop_samples_found = NA)
```

```{r}

#| label: eDNA-format-phylum-long-indirect

# Calculate number of samples which detected each species per site

## create pa variable
eDNA_phylum <- eDNA_phylum %>%
  mutate(pa = ifelse(reads > 0, 1, 0))

## Convert multiple columns to factor
eDNA_phylum <- eDNA_phylum %>%
  mutate(
    taxa = factor(taxa),
    localityID = factor(localityID),
    fieldID = factor(fieldID),
    sampleType = factor(sampleType),
    phylum = factor(phylum)
  )

## summarise to calculate proportions of detection at a site level
eDNA_site_level_phylum <- eDNA_phylum %>%
    dplyr::group_by(localityID, phylum) %>%  # Group by site and taxa
    dplyr::summarise(
        num_samples = n_distinct(fieldID),
        num_samples_found = sum(pa > 0, na.rm = TRUE),
        prop_samples_found = sum(pa > 0, na.rm = TRUE)/n_distinct(fieldID)
    ) %>%
    ungroup()

## rename columns
colnames(eDNA_site_level_phylum) <- c("localityID", "phylum", "num_samples", "num_samples_found", "prop_samples_found")

## get columns of interest
eDNA_subset_phylum <- eDNA_phylum %>% dplyr::select(c(
    valid_name,
    localityID,
    reads,
    country,
    phylum,
    verbatimLocality,
    pa
  ))

## rename columns
colnames(eDNA_subset_phylum) <- c("taxa",
                                 "localityID",
                                 "count",
                                 "region",
                                 "phylum",
                                 "site",
                                 "pa")

## add method
eDNA_subset_phylum$method <- "eDNA"

# Compute unique present taxa per phylum + site
num_taxa_summary_phylum <- eDNA_subset_phylum %>%
  filter(count > 0) %>%
  group_by(phylum, site) %>%
  summarise(num_taxa_unique = n_distinct(taxa), .groups = "drop")

# add observations from different species but same phylum together, cal richness
eDNA_subset_phylum <- eDNA_subset_phylum %>%
  group_by(phylum, site) %>%
  summarise(
    localityID = first(localityID),
    count = sum(count),
    region = first(region),
    method = first(method),
    .groups = 'drop'
  )

# Join back to original data (keeps absences)
eDNA_subset_phylum <- eDNA_subset_phylum %>%
  left_join(num_taxa_summary_phylum, by = c("phylum", "site")) %>% 
  mutate(num_taxa_unique = replace_na(num_taxa_unique, 0))

## join proportions and summed counts together 
eDNA_join_phylum <- left_join(eDNA_subset_phylum,
                              eDNA_site_level_phylum,
                              by = c("phylum", "localityID")) %>%
  mutate(pa = ifelse(count > 0, 1, 0))

## select relevant columns only for the final data
eDNA_join_final_phylum <- eDNA_join_phylum %>%  select(c(localityID, count, region, phylum, site, method, num_taxa_unique, pa, num_samples, num_samples_found, prop_samples_found))

## check pa and counts align
any(eDNA_join_final_phylum$num_samples_found > 0 & eDNA_join_final_phylum$count == 0) #should be false
```

```{r}

#| label: combine-methods-phylum-long-indirect

# check columns align - should all be true
colnames(visual_subset_phylum) == colnames(eDNA_join_final_phylum)

# combine
combined_data_phylum<- rbind(visual_subset_phylum, eDNA_join_final_phylum)

# remove any white space
combined_data_phylum$phylum <- trimws(combined_data_phylum$phylum)

# remove any duplicates
combined_data_phylum <- unique(combined_data_phylum)
```

We now have four version of combined datasets representing indirect matches.

```{r}

#| label: indirect-datasets-done

head(combined_data)
head(combined_data_genus)
head(combined_data_family)
head(combined_data_phylum)
```

### Combine (indirect, `phyloseq`)

Now let's make the combined `phyloseq` object.

Species-level.

```{r}

# correcting captials
otu_table(phylo_MarClim) <- round(otu_table(phylo_MarClim))
otu_table(phylo_rocky_eDNA) <- round(otu_table(phylo_rocky_eDNA))

taxa_names(phylo_MarClim) <- sapply(taxa_names(phylo_MarClim), function(x) {
  paste(toupper(substring(x, 1, 1)), tolower(substring(x, 2)), sep = "")
})

# phylo - match to species, genus, family and phylum
visual_family <- unique(visual$family)
visual_phyla <- unique(visual$phylum)

# species
phylo_rocky_eDNA_marlcim_species <- subset_taxa(phylo_rocky_eDNA,
                                        valid_name %in% visual_species)
phylo_rocky_eDNA_marlcim_species

# formatting eDNA phylo
phylo_eDNA_merge <- phylo_rocky_eDNA_marlcim_species
sample_data(phylo_eDNA_merge)$Method <- "eDNA"
sample_data(phylo_eDNA_merge)$methodsite_id <- paste(sample_data(phylo_eDNA_merge)$localityID, sample_data(phylo_eDNA_merge)$Method, sep = "_")

# reduce eDNA sample data
sd <- sample_data(phylo_eDNA_merge) %>% data.frame()
sd <- sd[, names(sd) %in% c("localityID", "country", "verbatimLocality", "Method", "methodsite_id")]
sd$localityID <- as.factor(sd$localityID)
sd$country <- as.factor(sd$country)
sd$Method <- as.factor(sd$Method)
sd$verbatimLocality <- as.factor(sd$verbatimLocality)
sd$methodsite_id <- as.factor(sd$methodsite_id)
rownames(sd) <- NULL
sd <- sd[!duplicated(sd), ]
rownames(sd) <- sd$methodsite_id

# merge DNA samples by sites
ps_edna_site <- merge_samples(phylo_eDNA_merge, group = "methodsite_id")
sample_data(ps_edna_site) <- sd

# formatting visual
phylo_visual_merge <- phylo_MarClim
sample_data(phylo_visual_merge)$Method <- "Visual"
sample_data(phylo_visual_merge)$methodsite_id <- paste(sample_data(phylo_visual_merge)$localityID, sample_data(phylo_visual_merge)$Method, sep = "_")
sample_names(phylo_visual_merge) <- sample_data(phylo_visual_merge)$methodsite_id

sample_data(phylo_visual_merge)$localityID <- as.factor(sample_data(phylo_visual_merge)$localityID)
sample_data(phylo_visual_merge)$country <- as.factor(sample_data(phylo_visual_merge)$country)
sample_data(phylo_visual_merge)$Method <- as.factor(sample_data(phylo_visual_merge)$Method)
sample_data(phylo_visual_merge)$verbatimLocality <- as.factor(sample_data(phylo_visual_merge)$verbatimLocality)
sample_data(phylo_visual_merge)$methodsite_id <- as.factor(sample_data(phylo_visual_merge)$methodsite_id)

# merge together
ps_merged <- merge_phyloseq(phylo_visual_merge, ps_edna_site)
ps_merged

ps_merged_pa <- microViz::tax_transform(ps_merged, trans = "pa")

# Remove any taxa without occurances
ps_merged_pa <- subset_taxa(ps_merged_pa, taxa_sums(ps_merged_pa) > 0)
```

Genus-level.

```{r}

# genus
phylo_rocky_eDNA_marlcim_genus <- subset_taxa(phylo_rocky_eDNA,
                                        genus %in% visual_genera)
phylo_rocky_eDNA_marlcim_genus

# formatting eDNA phylo
phylo_eDNA_merge_genus <- phylo_rocky_eDNA_marlcim_genus
sample_data(phylo_eDNA_merge_genus)$Method <- "eDNA"
sample_data(phylo_eDNA_merge_genus)$methodsite_id <- paste(sample_data(phylo_eDNA_merge_genus)$localityID, sample_data(phylo_eDNA_merge_genus)$Method, sep = "_")

# reduce eDNA sample data
sd_genus <- sample_data(phylo_eDNA_merge_genus) %>% data.frame()
sd_genus <- sd_genus[, names(sd_genus) %in% c("localityID", "country", "verbatimLocality", "Method", "methodsite_id")]
sd_genus$localityID <- as.factor(sd_genus$localityID)
sd_genus$country <- as.factor(sd_genus$country)
sd_genus$Method <- as.factor(sd_genus$Method)
sd_genus$verbatimLocality <- as.factor(sd_genus$verbatimLocality)
sd_genus$methodsite_id <- as.factor(sd_genus$methodsite_id)
rownames(sd_genus) <- NULL
sd_genus <- sd_genus[!duplicated(sd_genus), ]
rownames(sd_genus) <- sd_genus$methodsite_id

# merge DNA samples by sites
ps_edna_site_genus <- merge_samples(phylo_eDNA_merge_genus, group = "methodsite_id")
sample_data(ps_edna_site_genus) <- sd_genus

# merge together
ps_merged_genus <- merge_phyloseq(phylo_visual_merge, ps_edna_site_genus)
ps_merged_genus

ps_merged_genus_pa <- microViz::tax_transform(ps_merged_genus, trans = "pa")

# Remove any taxa without occurances
ps_merged_genus_pa <- subset_taxa(ps_merged_genus_pa, taxa_sums(ps_merged_genus_pa) > 0)
```

Family-level.

```{r}

# family
phylo_rocky_eDNA_marlcim_family <- subset_taxa(phylo_rocky_eDNA,
                                        family %in% visual_family)
phylo_rocky_eDNA_marlcim_family

# formatting eDNA phylo
phylo_eDNA_merge_family <- phylo_rocky_eDNA_marlcim_family
sample_data(phylo_eDNA_merge_family)$Method <- "eDNA"
sample_data(phylo_eDNA_merge_family)$methodsite_id <- paste(sample_data(phylo_eDNA_merge_family)$localityID, sample_data(phylo_eDNA_merge_family)$Method, sep = "_")

# reduce eDNA sample data
sd_family <- sample_data(phylo_eDNA_merge_family) %>% data.frame()
sd_family <- sd_family[, names(sd_family) %in% c("localityID", "country", "verbatimLocality", "Method", "methodsite_id")]
sd_family$localityID <- as.factor(sd_family$localityID)
sd_family$country <- as.factor(sd_family$country)
sd_family$Method <- as.factor(sd_family$Method)
sd_family$verbatimLocality <- as.factor(sd_family$verbatimLocality)
sd_family$methodsite_id <- as.factor(sd_family$methodsite_id)
rownames(sd_family) <- NULL
sd_family <- sd_family[!duplicated(sd_family), ]
rownames(sd_family) <- sd_family$methodsite_id

# merge DNA samples by sites
ps_edna_site_family <- merge_samples(phylo_eDNA_merge_family, group = "methodsite_id")
sample_data(ps_edna_site_family) <- sd_family

# merge together
ps_merged_family <- merge_phyloseq(phylo_visual_merge, ps_edna_site_family)
ps_merged_family

ps_merged_family_pa <- microViz::tax_transform(ps_merged_family, trans = "pa")

# Remove any taxa without occurances
ps_merged_family_pa <- subset_taxa(ps_merged_family_pa, taxa_sums(ps_merged_family_pa) > 0)
```

Phylum level

```{r}

# family
phylo_rocky_eDNA_marlcim_phylum <- subset_taxa(phylo_rocky_eDNA,
                                        phylum %in% visual_phylum)
phylo_rocky_eDNA_marlcim_phylum
```

### Combine (direct, long)

```{r}

#| label: aglomerated-data-long

eDNA_aglom <- eDNA %>% 
  subset(select = c(valid_name, localityID, verbatimLocality, reads, country)) %>% 
  mutate(Method = "eDNA")

visual_direct <- visual %>% 
  subset(select = c(valid_name, localityID, verbatimLocality, Count, country))%>% 
  mutate(Method = "Visual")
```

### Combine (direct, `phyloseq`)

```{r}

#| label: aglomerated-data-phylo

# aglom
phylo_MarClim_aglom_sp <- tax_glom(phylo_MarClim, taxrank="sciname")
phylo_MarClim_aglom_genus <- tax_glom(phylo_MarClim, taxrank="genus")
phylo_MarClim_aglom_family <- tax_glom(phylo_MarClim, taxrank="family")

phylo_rocky_eDNA_aglom_sp <- tax_glom(phylo_rocky_eDNA, taxrank="valid_name")
phylo_rocky_eDNA_aglom_genus <- tax_glom(phylo_rocky_eDNA, taxrank="genus")
phylo_rocky_eDNA_aglom_family <- tax_glom(phylo_rocky_eDNA, taxrank="family")

# filter eDNA by visual taxa
phylo_rocky_eDNA_aglom_sp <- subset_taxa(phylo_rocky_eDNA_aglom_sp,
                                        valid_name %in% visual_species)

phylo_rocky_eDNA_aglom_genus <- subset_taxa(phylo_rocky_eDNA_aglom_genus,
                                        genus  %in% visual_genera)

phylo_rocky_eDNA_aglom_family <- subset_taxa(phylo_rocky_eDNA_aglom_family,
                                        family %in% visual_family)

# merge eDNA by site
phylo_rocky_eDNA_aglom_sp <- merge_samples(phylo_rocky_eDNA_aglom_sp,
                                        "localityID")
# Remove any taxa without occurances
phylo_rocky_eDNA_aglom_sp <- subset_taxa(phylo_rocky_eDNA_aglom_sp, taxa_sums(phylo_rocky_eDNA_aglom_sp) > 0)

# merge eDNA by genus
phylo_rocky_eDNA_aglom_genus <- merge_samples(phylo_rocky_eDNA_aglom_genus,
                                        "localityID")
# Remove any taxa without occurances
phylo_rocky_eDNA_aglom_genus <- subset_taxa(phylo_rocky_eDNA_aglom_genus, taxa_sums(phylo_rocky_eDNA_aglom_genus) > 0)

# merge eDNA by family
phylo_rocky_eDNA_aglom_family <- merge_samples(phylo_rocky_eDNA_aglom_family,
                                        "localityID")
# Remove any taxa without occurances
phylo_rocky_eDNA_aglom_family <- subset_taxa(phylo_rocky_eDNA_aglom_family, taxa_sums(phylo_rocky_eDNA_aglom_family) > 0)
```

## Map of sampling sites

```{r}

#| label: get-map-elements

#blank map
worldmap = map_data('world')
worldUk <- subset(worldmap, region == c('UK','Ireland'))

#convert to sf object
sf_data <- st_as_sf(worldUk, coords = c("long", "lat"), crs = 4326)  # Specify CRS as WGS84 (EPSG:4326)

# Get countries outline
UK <- rnaturalearth::ne_countries(
  country = c("United Kingdom", "Ireland"),
  scale = "large", returnclass = "sf"
)
```

```{r}

#| label: get-map-lat-longs
 
unique_data <- eDNA %>% distinct(decimalLongitude, decimalLatitude, verbatimLocality, .keep_all = TRUE)

# Define custom colors for each country
country_colors <- c("Scotland" = "darkseagreen3",
                    "North Wales" = "lightpink2",
                    "Northeast England" = "lightyellow2",
                    "South Wales" = "lightblue2",
                    "Southwest England" = "plum")
```

```{r}
#| label: build-map

site_map <- ggplot() + 
  geom_polygon(data = worldUk, 
               aes(x = long, y = lat, group = group), 
               fill = 'grey97', color = 'black') + 
  coord_fixed(ratio = 1.3, xlim = c(-10, 3), ylim = c(50, 59)) + 
  theme_classic() + 
  scale_x_continuous(sec.axis = dup_axis()) +
  scale_y_continuous(sec.axis = dup_axis()) +
  geom_point(data = unique_data, 
             aes(x = as.numeric(decimalLongitude), 
                 y = as.numeric(decimalLatitude), 
                 color = country), 
             size = 2, alpha = .7) +  # Adjusted size for better visibility
  geom_label_repel(data = unique_data, 
                   aes(x = as.numeric(decimalLongitude), 
                       y = as.numeric(decimalLatitude), 
                       label = verbatimLocality,
                       fill = country),  # Using 'verbatimLocality' as the label
                   size = 2, 
                   segment.size = 0.7,  # Adjust segment size
                   box.padding = 0.3,  # Adjust padding around labels
                   label.size = 0.25,
                   max.overlaps = 100) +  # Border size of the label
  scale_fill_manual(values = country_colors, guide = "none") +  # Apply custom colors and remove fill legend
  scale_color_discrete(name = "Region") +  # Changed legend name to 'Region'
  theme(legend.position = "bottom",
        axis.title = element_blank(),
        axis.line = element_line(colour = "grey40",
                                 linewidth = 0.8),
        axis.text = element_text(colour = "grey40")) +
  ggspatial::annotation_north_arrow(
    location = "br", which_north = "true",
    #pad_x = unit(0.4, "in"), pad_y = unit(0.4, "in"),
    style = ggspatial::north_arrow_nautical(
      fill = c("grey40", "white"),
      line_col = "grey20",
      text_family = "ArcherPro Book"
    )
  ) +
  ggspatial::annotation_scale(
    location = "tr",
    bar_cols = c("grey60", "white"),
    text_family = "ArcherPro Book"
  ) 

ggsave(filename = "Figures/site_map_chpt2.png", plot = site_map,
       device = "png", dpi = 300, units = "mm", height = 180, width = 190)

site_map
```

## Detection concordance from paired sampling

### Bubble plots

We want to visualize our detection across sites and phyla.

First, we need to set-up some variables for the plots.

```{r}

#| label: set-variables

site_order <- c(
  "Scourie",
  "Rispond",
  "Skerray",
  "Murkle Bay",
  "Portskerra",
  #"Borwick, Yesnaby",
  "Sannick",
  "Wick", #Scotland
  "Great Orme East",
  "Little Orme",
  "Menai Bridge",
  "Porth Oer",
  "Porth Swtan",
  "Rhosneigr", #North Wales
  "Castlehead Rocks",
  "Filey Brigg",
  "Newton Point",
  "Rumbling Kern",
  "Scalby Mills", #Northumbria
  "Aberystwyth",
  "Neyland",
  "Broad Haven",
  #"Skomer Island",
  "West Angle",
  "Monkstone Point",
  "Dale Jetty",
  #"Martin's Haven", #South Wales
  "Lizard Point",
  "Looe",
  "Sennen Cove",
  "St Ives",
  "Trevone" #Cornwall
) 

country_order <- as.factor(c(
  "Scotland",
  "Northeast England",
  "North Wales",
  "South Wales",
  "Southwest England"
))

colours = c(
  "Scotland" = "darkseagreen1",
  "North Wales" = "lightpink",
  "Northeast England" = "lightyellow",
  "South Wales" = "lightblue1",
  "Southwest England" = "lavender"
)

```

Species-level bubble plot.

```{r}

#| label: plot-comp-bubble-species

# remove those note detected throughout the whole dataset
combined_data_matched <- combined_data %>%
  #filter(!genus == taxa) %>% # use this to remove codium and mytilus
  dplyr::group_by(taxa) %>% 
  filter(sum(pa) > 0) %>%    # Keep only taxa with at least one pa == 1
  ungroup()

# correct order for region
combined_data_matched$region <- factor(combined_data_matched$region, 
                                       levels = c("Scotland", 
                                                  "Northeast England",
                                                  "North Wales",
                                                  "South Wales",
                                                  "Southwest England"))

# check min and max of count
visual_data_check <- subset(combined_data_matched, method == "Visual" & count > 0)
min(visual_data_check$count)
max(visual_data_check$count)
#histogram(visual_data_check$count)

# Plot
comp_bubble_species <- ggplot(combined_data_matched, aes(x = site, y = taxa)) +
  # eDNA detections (size and fill mapped to prop_samples_found)
  geom_point(aes(
    size = if_else(method == "eDNA" & pa == 1, 3, NA_real_),
    fill = if_else(method == "eDNA" & pa == 1, prop_samples_found, NA_real_)
  ),
  alpha = 0.75,
  shape = 22) +

  # Visual detections (size applied, but using color instead of fill)
  geom_point(aes(
    size = if_else(method == "Visual" & pa == 1, count/8, NA_real_),
    color = if_else(method == "Visual" & pa == 1, "red", NA_character_)
  ), shape = 21) +

  # Labels
  labs(x = "",
       y = "",
       size = "",
       fill = "Proportion of samples with positive detection") +

  # Theme settings
  theme(
    legend.key = element_blank(),
    axis.text.x = element_text(
      colour = "black",
      size = 12,
      angle = 90,
      vjust = 0.3,
      hjust = 1
    ),
    axis.text.y = element_text(
      colour = "black",
      size = 11,
      face = "italic"
    ),
    legend.text = element_text(size = 10, colour = "black"),
    legend.title = element_text(size = 10),
    panel.background = element_blank(),
    panel.border = element_rect(
      colour = "black",
      fill = NA,
      size = 1.2
    ),
    legend.position = "none",
    strip.text.y = element_text(size = 6),
    strip.text.x = element_text(size = 6)
  ) +

  # Use a continuous fill scale for eDNA values
  scale_fill_gradient(
    low = "honeydew2",
    high = "darkgreen",
    na.value = "grey"
  ) +

  # Use a manual color scale for visual detections
  scale_color_manual(values = c("red" = "red"), na.translate = FALSE) +

  # Faceting
  facet_grid(phylum ~ region, scales = "free", space = "free")

comp_bubble_species
```

Genus-level bubble plot.

```{r}
#| label: plot-comp-bubble-genus

# remove taxa without any detections across the dataset
combined_data_genus <- combined_data_genus %>%
  dplyr::group_by(genus) %>% 
  filter(sum(pa) > 0) %>%    # Keep only taxa with at least one pa == 1
  ungroup()

combined_data_genus$region <- factor(combined_data_genus$region, 
                                       levels = c("Scotland", 
                                                  "Northeast England",
                                                  "North Wales",
                                                  "South Wales",
                                                  "Southwest England"))

# Plot
comp_bubble_genus <- ggplot(combined_data_genus, aes(x = site, y = genus)) +
  # eDNA detections (size and fill mapped to prop_samples_found)
  geom_point(aes(
    size = if_else(method == "eDNA" & pa == 1, 3, NA_real_),
    fill = if_else(method == "eDNA" & pa == 1, prop_samples_found, NA_real_)
  ),
  alpha = 0.75,
  shape = 22) +

  # Visual detections (size applied, but using color instead of fill)
  geom_point(aes(
    size = if_else(method == "Visual" & pa == 1, 1, NA_real_),
    color = if_else(method == "Visual" & pa == 1, "red", NA_character_)
  ), shape = 21) +

  # Labels
  labs(x = "",
       y = "",
       size = "",
       fill = "Proportion of samples with positive detection") +

  # Theme settings
  theme(
    legend.key = element_blank(),
    axis.text.x = element_text(
      colour = "black",
      size = 12,
      angle = 90,
      vjust = 0.3,
      hjust = 1
    ),
    axis.text.y = element_text(
      colour = "black",
      size = 11,
      face = "italic"
    ),
    legend.text = element_text(size = 10, colour = "black"),
    legend.title = element_text(size = 10),
    panel.background = element_blank(),
    panel.border = element_rect(
      colour = "black",
      fill = NA,
      size = 1.2
    ),
    legend.position = "none",
    strip.text.y = element_text(size = 6),
    strip.text.x = element_text(size = 6)
  ) +

  # Use a continuous fill scale for eDNA values
  scale_fill_gradient(
    low = "lightblue1",
    high = "deepskyblue4",
    na.value = "grey"
  ) +

  # Use a manual color scale for visual detections
  scale_color_manual(values = c("red" = "red"), na.translate = FALSE) +

  # Faceting
  facet_grid(phylum ~ region, scales = "free", space = "free")

comp_bubble_genus
```

Family-level bubble plot.

```{r}
#| label: plot-comp-bubble-family

# remove taxa without any detections across the dataset
combined_data_family <- combined_data_family %>%
  dplyr::group_by(family) %>% 
  filter(sum(pa) > 0) %>%    # Keep only taxa with at least one pa == 1
  ungroup()

combined_data_family$region <- factor(combined_data_family$region, 
                                       levels = c("Scotland", 
                                                  "Northeast England",
                                                  "North Wales",
                                                  "South Wales",
                                                  "Southwest England"))

# Plot
comp_bubble_family <- ggplot(combined_data_family, aes(x = site, y = family)) +
  # eDNA detections (size and fill mapped to prop_samples_found)
  geom_point(aes(
    size = if_else(method == "eDNA" & pa == 1, 3, NA_real_),
    fill = if_else(method == "eDNA" & pa == 1, prop_samples_found, NA_real_)
  ),
  alpha = 0.75,
  shape = 22) +

  # Visual detections (size applied, but using color instead of fill)
  geom_point(aes(
    size = if_else(method == "Visual" & pa == 1, 1, NA_real_),
    color = if_else(method == "Visual" & pa == 1, "red", NA_character_)
  ), shape = 21) +

  # Labels
  labs(x = "",
       y = "",
       size = "",
       fill = "Proportion of samples with positive detection") +

  # Theme settings
  theme(
    legend.key = element_blank(),
    axis.text.x = element_text(
      colour = "black",
      size = 12,
      angle = 90,
      vjust = 0.3,
      hjust = 1
    ),
    axis.text.y = element_text(
      colour = "black",
      size = 11,
      face = "italic"
    ),
    legend.text = element_text(size = 10, colour = "black"),
    legend.title = element_text(size = 10),
    panel.background = element_blank(),
    panel.border = element_rect(
      colour = "black",
      fill = NA,
      size = 1.2
    ),
    legend.position = "none",
    strip.text.y = element_text(size = 6),
    strip.text.x = element_text(size = 6)
  ) +

  # Use a continuous fill scale for eDNA values
  scale_fill_gradient(
    low = "antiquewhite",
    high = "goldenrod4",
    na.value = "grey"
  ) +

  # Use a manual color scale for visual detections
  scale_color_manual(values = c("red" = "red"), na.translate = FALSE) +

  # Faceting
  facet_grid(phylum ~ region, scales = "free", space = "free")

comp_bubble_family
```

### Save bubble plots

```{r}

#| label: save-comp-bubbles

ggsave(filename = "Figures/comparison_bubble_species.png",
       plot = comp_bubble_species, device = "png", dpi = 300, units = "mm", height = 320, width = 200)

ggsave(filename = "Figures/comparison_bubble_genus.png",
       plot = comp_bubble_genus, device = "png", dpi = 300, units = "mm", height = 300, width = 180)

ggsave(filename = "Figures/comparison_bubble_family.png",
       plot = comp_bubble_family, device = "png", dpi = 300, units = "mm", height = 260, width = 180)
```

### Concordance table

We want to know the percentage match between the two methods across the sites. This is also done across species, genus and family.

```{r}

#| label: percentage-table-species

percentage_match <- combined_data %>%
  select(-count) %>%
  filter(pa == 1) %>%
  unique() %>%
  dplyr::group_by(taxa, region) %>%
  dplyr::summarise(
    eDNA_detected = sum(method == "eDNA"),
    # Count of eDNA detections for each taxa
    visual_detected = sum(method == "Visual"),
    # Count of visual detections for each taxa
    total_detected = eDNA_detected + visual_detected,
    # Total detections
    perc_match = ifelse(total_detected > 0, (eDNA_detected / total_detected) * 100, 0),
    # Calculate percentage match
    more_detected = case_when(
      eDNA_detected > visual_detected ~ "eDNA",
      visual_detected > eDNA_detected ~ "Visual",
      TRUE ~ "Equal"  # In case of a tie
    )
  ) %>%
  dplyr::ungroup()

percentage_match

# save as CSV
write.csv(percentage_match, file = "Processed_Data/percentage_match_species.csv", row.names = FALSE)

# not over regions
percentage_match_no_region <- combined_data %>%
  select(-count) %>%
  filter(pa == 1) %>%
  unique() %>%
  dplyr::group_by(taxa) %>%
  dplyr::summarise(
    eDNA_detected = sum(method == "eDNA"),
    # Count of eDNA detections for each taxa
    visual_detected = sum(method == "Visual"),
    # Count of visual detections for each taxa
    total_detected = eDNA_detected + visual_detected,
    # Total detections
    perc_match = ifelse(total_detected > 0, (eDNA_detected / total_detected) * 100, 0),
    # Calculate percentage match
    more_detected = case_when(
      eDNA_detected > visual_detected ~ "eDNA",
      visual_detected > eDNA_detected ~ "Visual",
      TRUE ~ "Equal"  # In case of a tie
    )
  ) %>%
  dplyr::ungroup()

# save as CSV
write.csv(percentage_match_no_region, file = "Processed_Data/percentage_match_species_no_region.csv", row.names = FALSE)
```

```{r}

#| label: percentage-table-genus

percentage_match_genus <- combined_data_genus %>%
  select(-count) %>%
  filter(pa == 1) %>%
  unique() %>%
  dplyr::group_by(genus, region) %>%
  dplyr::summarise(
    eDNA_detected = sum(method == "eDNA"),
    # Count of eDNA detections for each taxa
    visual_detected = sum(method == "Visual"),
    # Count of visual detections for each taxa
    total_detected = eDNA_detected + visual_detected,
    # Total detections
    perc_match = ifelse(total_detected > 0, (eDNA_detected / total_detected) * 100, 0),
    # Calculate percentage match
    more_detected = case_when(
      eDNA_detected > visual_detected ~ "eDNA",
      visual_detected > eDNA_detected ~ "Visual",
      TRUE ~ "Equal"  # In case of a tie
    )
  ) %>%
  dplyr::ungroup()

print(percentage_match_genus)

# save as CSV
write.csv(percentage_match_genus, file = "Processed_Data/percentage_match_genus.csv", row.names = FALSE)

# not over regions
percentage_match_genus_no_region <- combined_data_genus %>%
  select(-count) %>%
  filter(pa == 1) %>%
  unique() %>%
  dplyr::group_by(genus) %>%
  dplyr::summarise(
    eDNA_detected = sum(method == "eDNA"),
    # Count of eDNA detections for each taxa
    visual_detected = sum(method == "Visual"),
    # Count of visual detections for each taxa
    total_detected = eDNA_detected + visual_detected,
    # Total detections
    perc_match = ifelse(total_detected > 0, (eDNA_detected / total_detected) * 100, 0),
    # Calculate percentage match
    more_detected = case_when(
      eDNA_detected > visual_detected ~ "eDNA",
      visual_detected > eDNA_detected ~ "Visual",
      TRUE ~ "Equal"  # In case of a tie
    )
  ) %>%
  dplyr::ungroup()

# save as CSV
write.csv(percentage_match_genus_no_region, file = "Processed_Data/percentage_match_genus_no_region.csv", row.names = FALSE)
```

```{r}

#| label: percentage-table-family

percentage_match_family <- combined_data_family %>%
  select(-count) %>%
  filter(pa == 1) %>%
  unique() %>%
  dplyr::group_by(family, region) %>%
  dplyr::summarise(
    eDNA_detected = sum(method == "eDNA"),
    # Count of eDNA detections for each taxa
    visual_detected = sum(method == "Visual"),
    # Count of visual detections for each taxa
    total_detected = eDNA_detected + visual_detected,
    # Total detections
    perc_match = ifelse(total_detected > 0, (eDNA_detected / total_detected) * 100, 0),
    # Calculate percentage match
    more_detected = case_when(
      eDNA_detected > visual_detected ~ "eDNA",
      visual_detected > eDNA_detected ~ "Visual",
      TRUE ~ "Equal"  # In case of a tie
    )
  ) %>%
  dplyr::ungroup()

print(percentage_match_family)

# save as CSV
write.csv(percentage_match_family, file = "Processed_Data/percentage_match_family.csv", row.names = FALSE)

# not over regions
percentage_match_family_no_region <- combined_data_family %>%
  select(-count) %>%
  filter(pa == 1) %>%
  unique() %>%
  dplyr::group_by(family) %>%
  dplyr::summarise(
    eDNA_detected = sum(method == "eDNA"),
    # Count of eDNA detections for each taxa
    visual_detected = sum(method == "Visual"),
    # Count of visual detections for each taxa
    total_detected = eDNA_detected + visual_detected,
    # Total detections
    perc_match = ifelse(total_detected > 0, (eDNA_detected / total_detected) * 100, 0),
    # Calculate percentage match
    more_detected = case_when(
      eDNA_detected > visual_detected ~ "eDNA",
      visual_detected > eDNA_detected ~ "Visual",
      TRUE ~ "Equal"  # In case of a tie
    )
  ) %>%
  dplyr::ungroup()

# save as CSV
write.csv(percentage_match_family_no_region, file = "Processed_Data/percentage_match_family_no_region.csv", row.names = FALSE)
```

## Diversity

### Alpha diversity

Time to do our alpha diversity visualizations and analyses.

#### Calculate richness

Let's calculate richness at each taxonomic level for direct and indirect comparisons.

Indirect first.

```{r}

#| label: calculate-richness-species

# phyloseq
alpha_df_visual <- phyloseq::estimate_richness(physeq = phylo_MarClim, measure = c("Observed", "Shannon"))

alpha_df_eDNA <- phyloseq::estimate_richness(physeq = phylo_rocky_eDNA_marlcim_species, measure = c("Observed", "Shannon"))

# long
diversity_data <- combined_data %>%
  dplyr::group_by(region, method, site) %>%  
  dplyr::summarize(
    richness = sum(pa),  
    shannon = vegan::diversity(count, index = "shannon"),
    rank = "Species"
  ) %>%
  na.omit()%>% 
  mutate(method_rank = paste(rank, method, sep = "_"))

head(diversity_data)
```

```{r}

#| label: calculate-richness-genus

alpha_df_eDNA_genus <- phyloseq::estimate_richness(physeq = phylo_rocky_eDNA_marlcim_genus, measure = c("Observed", "Shannon"))

diversity_data_genus <- combined_data_genus %>%
  dplyr::group_by(region, method, site) %>%  
  dplyr::summarize(
    richness = sum(pa),  
    shannon = vegan::diversity(count, index = "shannon"),
    rank = "Genus"
  ) %>%
  na.omit()%>% 
  mutate(method_rank = paste(rank, method, sep = "_"))

head(diversity_data_genus)
```

```{r}

#| label: calculate-richness-family

alpha_df_eDNA_family <- phyloseq::estimate_richness(physeq = phylo_rocky_eDNA_marlcim_family, measure = c("Observed", "Shannon"))

diversity_data_family <- combined_data_family %>%
  dplyr::group_by(region, method, site) %>%  
  dplyr::summarize(
    richness = sum(pa),  
    shannon = vegan::diversity(count, index = "shannon"),
    rank = "Family") %>%
  na.omit() %>% 
  mutate(method_rank = paste(rank, method, sep = "_"))

head(diversity_data_family)
```

Phylum level

```{r}

#| label: calculate-richness-phylum

alpha_df_eDNA_phylum <- phyloseq::estimate_richness(physeq = phylo_rocky_eDNA_marlcim_phylum, measure = c("Observed", "Shannon"))

diversity_data_phylum <- combined_data_phylum %>%
  dplyr::group_by(region, method, site) %>%  
  dplyr::summarize(
    richness = sum(pa),  
    shannon = vegan::diversity(count, index = "shannon"),
    rank = "Phylum") %>%
  na.omit() %>% 
  mutate(method_rank = paste(rank, method, sep = "_"))

head(diversity_data_phylum)
```

```{r}

#| label: join-diversity-overall

diversity_overall <- rbind(diversity_data, diversity_data_genus, diversity_data_family, diversity_data_phylum)

#diversity_overall <- diversity_overall %>% 
  #subset(!method_rank %in% c("Family_Visual", "Genus_Visual"))
```

And let's calculate difference in richness between the two methods for plots and models later.

```{r}

# Duplicate Visual data for Genus and Family
#visual_expanded <- diversity_overall %>%
  #filter(method == "Visual") %>%
  #mutate(rank = "Genus") %>%
  #bind_rows(
    #diversity_overall %>% filter(method == "Visual") %>% mutate(rank = "Family")
  #)

#ranks <- c("Species", "Genus", "Family", "Phylum")
#diversity_overall_expanded <- bind_rows(diversity_overall, visual_expanded) %>%
  #filter(rank %in% ranks)  # Ensure only selected ranks

# make wide for scatter and cal difference
diversity_indirect_diff <- diversity_overall %>%
  select(site, rank, method, richness, region) %>%
  pivot_wider(names_from = method, values_from = richness) %>%
  mutate(richness_diff =Visual - `eDNA`)

diversity_indirect_diff$region = as.factor(diversity_indirect_diff$region)
diversity_indirect_diff$rank = as.factor(diversity_indirect_diff$rank)
diversity_indirect_diff$site = as.factor(diversity_indirect_diff$site)

# Reorder region factor levels
diversity_indirect_diff$region <- factor(diversity_indirect_diff$region, levels = country_order)
```

Direct next (using the agglomerated data).

```{r}

#| label: calculate-richness-aglomerated-data

# species - visual
visual_aglom_rich_sp <- phyloseq::estimate_richness(physeq = phylo_MarClim_aglom_sp, measure = "Observed")
visual_aglom_rich_sp$tax_rank = "Species"
visual_aglom_rich_sp$method = "Visual"
visual_aglom_rich_sp <- rownames_to_column(visual_aglom_rich_sp, var = "site")

# genus - visual
visual_aglom_rich_genus <- phyloseq::estimate_richness(physeq = phylo_MarClim_aglom_genus, measure = "Observed")
visual_aglom_rich_genus$tax_rank = "Genus"
visual_aglom_rich_genus$method = "Visual"
visual_aglom_rich_genus <- rownames_to_column(visual_aglom_rich_genus, var = "site")

# family - visual
visual_aglom_rich_family <- phyloseq::estimate_richness(physeq = phylo_MarClim_aglom_family, measure = "Observed")
visual_aglom_rich_family$tax_rank = "Family"
visual_aglom_rich_family$method = "Visual"
visual_aglom_rich_family <- rownames_to_column(visual_aglom_rich_family, var = "site")

# species - eDNA
eDNA_aglom_rich_sp <- phyloseq::estimate_richness(physeq = phylo_rocky_eDNA_aglom_sp, measure = "Observed")
eDNA_aglom_rich_sp$tax_rank = "Species"
eDNA_aglom_rich_sp$method = "eDNA"
eDNA_aglom_rich_sp <- rownames_to_column(eDNA_aglom_rich_sp, var = "site")

# genus - eDNA
eDNA_aglom_rich_genus <- phyloseq::estimate_richness(physeq = phylo_rocky_eDNA_aglom_genus, measure = "Observed")
eDNA_aglom_rich_genus$tax_rank = "Genus"
eDNA_aglom_rich_genus$method = "eDNA"
eDNA_aglom_rich_genus <- rownames_to_column(eDNA_aglom_rich_genus, var = "site")

# family - eDNA
eDNA_aglom_rich_family <- phyloseq::estimate_richness(physeq = phylo_rocky_eDNA_aglom_family, measure = "Observed")
eDNA_aglom_rich_family$tax_rank = "Family"
eDNA_aglom_rich_family$method = "eDNA"
eDNA_aglom_rich_family <- rownames_to_column(eDNA_aglom_rich_family, var = "site")

# format to join together
diversity_agglom <- rbind(visual_aglom_rich_sp, visual_aglom_rich_genus, visual_aglom_rich_family, eDNA_aglom_rich_sp, eDNA_aglom_rich_genus, eDNA_aglom_rich_family)

# add region back in
metadata <- sample_data(phylo_MarClim_aglom_sp)
colnames(metadata)[1] <- "site"
diversity_agglom <- left_join(diversity_agglom, metadata)
```

Now let's calculate the difference in richness.

```{r}

#| label: calculate-richness-direct-difference

# make wide for simple box plots
diversity_agglom_diff <- diversity_agglom %>%
  pivot_wider(names_from = method, values_from = Observed) %>%
  mutate(diff_Observed = Visual - eDNA)

# Reorder region factor levels
diversity_agglom_diff$country <- factor(diversity_agglom_diff$country, levels = country_order)
```

#### Models

Direct richness model first.

```{r}

#| label: direct-richness-model

# visualise
histogram(diversity_agglom_diff$diff_Observed) # very slight right skewed

# construct model
richess_dirct_mod <- lm(
  diff_Observed ~ tax_rank * country,
  data = diversity_agglom_diff
)

# checks
plot(richess_dirct_mod) #fine

# outputs
summary(richess_dirct_mod)
Anova(richess_dirct_mod, type = "II")

# means
summary_direct <- group_by(diversity_agglom_diff, tax_rank) %>%
  summarise(
    mean = mean(diff_Observed, na.rm = TRUE),
    sd = sd(diff_Observed, na.rm = TRUE)
  )

summary_direct
```

And indirect richness model.

```{r}

#| label: indirect-richness-model

# visualise
histogram(diversity_indirect_diff$richness_diff)

# construct model
richess_indirct_mod <- lm(
  richness_diff ~ rank * region,
  data = diversity_indirect_diff
)

# checks
plot(richess_indirct_mod) #fine

# outputs
summary(richess_indirct_mod)
Anova(richess_indirct_mod, type = "II")

# means
summary_indirect <- group_by(diversity_indirect_diff, rank) %>%
  summarise(
    mean = mean(richness_diff, na.rm = TRUE),
    sd = sd(richness_diff, na.rm = TRUE)
  )

summary_indirect
```

#### Plots

We are going to make a few different alpha plots using indirect and direct comparisons.

Indirect matches plots.

```{r}

#| label: plot-diversity-overall-box

rank_order <- c("Species_Visual", "Species_eDNA", "Genus_eDNA","Family_eDNA")

richness_comp_overall <- ggplot(diversity_overall,
                                aes(x = method_rank, y = richness, fill = method_rank)) +
  geom_boxplot() +
  scale_fill_manual(values = c(
    "Family_eDNA" = "deepskyblue2",
    "Genus_eDNA" = "deepskyblue3",
    "Species_eDNA" = "deepskyblue4",
    "Species_Visual" = "darkseagreen"
  )) + 
  scale_x_discrete(labels = c(
    "Family_eDNA" = "Families (eDNA)",
    "Genus_eDNA" = "Genera (eDNA)",
    "Species_eDNA" = "Species (eDNA)",
    "Species_Visual" = "Species (Visual)"
  ), limits = rank_order) +
  geom_jitter(
    color = "black",
    size = 0.9,
    alpha = 0.9,
    width = 0.1,
    height = 0.1
  ) +
  labs(x = "Matched taxonomic rank and method", y = "Number of taxa detected within taxonomic rank") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none")

richness_comp_overall_region <- ggplot(diversity_overall,
                                aes(x = method_rank, y = richness, fill = method_rank)) +
  geom_boxplot() +
  scale_fill_manual(values = c(
    "Family_eDNA" = "deepskyblue2",
    "Genus_eDNA" = "deepskyblue3",
    "Species_eDNA" = "deepskyblue4",
    "Species_Visual" = "darkseagreen"
  )) + 
  scale_x_discrete(labels = c(
    "Family_eDNA" = "Families (eDNA)",
    "Genus_eDNA" = "Genera (eDNA)",
    "Species_eDNA" = "Species (eDNA)",
    "Species_Visual" = "Species (Visual)"
  ), limits = rank_order) +
  labs(x = "Matched taxonomic rank and method", y = "Number of taxa detected within taxonomic rank") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none") +
  facet_wrap(~ region, ncol = 5)

alpha_diversity_overall_grid <- cowplot::plot_grid(richness_comp_overall, richness_comp_overall_region, labels = c("a", "b"), ncol = 1)

# save
ggsave(filename = "Figures/alpha_comp_plot_overall.png",
       plot = alpha_diversity_overall_grid, device = "png", dpi = 300, units = "mm", height = 250, width = 200)

alpha_diversity_overall_grid
```

Scatter plot to show difference in indirect richness across ranks.

```{r}

# mod
method_lm <- lm(eDNA ~ Visual * rank, diversity_indirect_diff)
summary(method_lm)
plot(method_lm)

anova(method_lm)

# Extract the prediction data frame
pred.indirect <- ggeffects::ggpredict(method_lm, terms = c("Visual", "rank"))

# scatter
scatter_indirect <- ggplot(diversity_indirect_diff, aes(x = Visual, y = eDNA, colour = rank)) +
  geom_point() +
  
  # Predicted lines (suppress global aes)
  geom_line(
    data = pred.indirect,
    aes(x = x, y = predicted, colour = group),
    linewidth = 1.2,
    inherit.aes = FALSE
  ) +
  
  # Confidence ribbon (suppress global aes)
  geom_ribbon(
    data = pred.indirect,
    aes(
      x = x,
      ymin = predicted - std.error,
      ymax = predicted + std.error,
      fill = group
    ),
    alpha = 0.3,
    inherit.aes = FALSE
  ) +
  
  geom_abline(intercept = 0, slope = 1, linetype = "dashed", colour = "black") +
  
  scale_colour_manual(values = c("Species" = "#F6AF65", 
                                 "Genus" = "#00C2D1",
                                 "Family" = "#F9E108")) +
    scale_fill_manual(values = c("Species" = "#F6AF65", 
                                 "Genus" = "#00C2D1",
                                 "Family" = "#F9E108")) +
  labs(colour = "Taxonomic rank", 
       fill = "Taxonomic rank",
       x = "Visual richness",
       y = "eDNA richness") +
  theme_classic() 

# save
ggsave(filename = "Figures/alpha_indirect_satter.png",
       plot = scatter_indirect, device = "png", dpi = 300, units = "mm", height = 120, width = 170)

scatter_indirect
```

Now let's make the direct matches plot.

```{r}

#| label: plot-diversity-agglom-box

diversity_agglom$tax_rank <- factor(diversity_agglom$tax_rank, levels = c("Species", "Genus", "Family"))

richness_comp_aglom <- ggplot(diversity_agglom, aes(x = country, y = Observed, fill = method)) +
  geom_boxplot() +
  labs(x = "Region", y = "Richness", pattern = "Survey method") +
  scale_fill_manual(values = c("eDNA" = "deepskyblue4", 
                               "Visual" = "darkseagreen"))+
  theme_bw() +
  scale_x_discrete(limits = country_order) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "bottom") +
  labs(fill = "Method")+
  facet_grid(~ tax_rank)

richness_comp_overall_aglom <- ggplot(diversity_agglom, aes(x = method, y = Observed, fill = method)) +
  geom_boxplot()+
  geom_line(aes(group=site), colour = "grey50") +
  scale_fill_manual(values = c("eDNA" = "deepskyblue4", 
                               "Visual" = "darkseagreen"))+
  geom_jitter(color="black", size=0.9, alpha=0.9, width = 0.1, height = 0.1) +
  labs(x = "Method", y = "Richness") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none")+
  facet_grid(~ tax_rank)

#plot together
alpha_diversity_overall_grid_aglom <- cowplot::plot_grid(richness_comp_overall_aglom,richness_comp_aglom, labels = c("a", "b"), ncol = 1, rel_heights = c(1,1.2))
```

Save direct comparison plot.

```{r}

#| label: save-diversity-agglom-box

ggsave(filename = "Figures/alpha_comp_plot_aglom.png",
       plot = alpha_diversity_overall_grid_aglom, device = "png", dpi = 300, units = "mm", height = 230, width = 200)

alpha_diversity_overall_grid_aglom
```

Plots for difference in richness.

```{r}

#| label: direct-alpha-diversity-boxplots

tax_order <- c("Species", "Genus", "Family")

# Basic boxplot with dashed line at 0
simple_box_direct <- ggplot(diversity_agglom_diff, aes(x = tax_rank, y = diff_Observed, fill = tax_rank)) +
  geom_boxplot() +
  geom_hline(yintercept = 0, linetype = "dashed", color = "black") +  
  labs(x = "Taxonomic rank", y = "Difference in richness") +
  scale_fill_manual(values = c("Species" = "#F6AF65", 
                               "Genus" = "#00C2D1",
                               "Family" = "#F9E900")) +
  theme_bw() +
  scale_x_discrete(limits = tax_order) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none") +
  labs(fill = "Taxonomic rank")

# Faceted boxplot with dashed line at 0
simple_box_direct_region <- ggplot(diversity_agglom_diff, aes(x = tax_rank, y = diff_Observed, fill = tax_rank)) +
  geom_boxplot() +
  geom_hline(yintercept = 0, linetype = "dashed", color = "black") +  
  labs(x = "Taxonomic rank", y = "Difference in richness") +
  scale_fill_manual(values = c("Species" = "#F6AF65", 
                               "Genus" = "#00C2D1",
                               "Family" = "#F9E900")) +
  theme_bw() +
  scale_x_discrete(limits = tax_order) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none") +
  labs(fill = "Taxonomic rank") +
  facet_wrap(~ country, ncol = 5)

# plot together
alpha_diversity_direct_diff <- cowplot::plot_grid(simple_box_direct,simple_box_direct_region, labels = c("a", "b"), ncol = 1)

ggsave(filename = "Figures/alpha_comp_plot_aglom_diff.png",
       plot = alpha_diversity_direct_diff, device = "png", dpi = 300, units = "mm", height = 230, width = 170)

alpha_diversity_direct_diff
```

### Beta diversity

#### Composition plots

```{r}

#| label: composition-diversity-plots-species

# Compute relative abundance
df_rel <- combined_data %>%
  group_by(region, method, phylum) %>%
  summarise(total_count = sum(count), .groups = "drop") %>%
  group_by(region, method) %>%
  mutate(rel_abundance = total_count / sum(total_count))

# Stacked bar plot
species_stacked <- ggplot(df_rel, aes(x = method, y = rel_abundance, fill = phylum)) +
  geom_bar(stat = "identity", position = "stack") +
  facet_wrap(~region, ncol = 5) +
  labs(x = "Method", y = "Relative Abundance", fill = "Phylum") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none")
```

```{r}

#| label: composition-diversity-plots-genus

# Compute relative abundance
df_rel <- combined_data_genus %>%
  group_by(region, method, phylum) %>%
  summarise(total_count = sum(count), .groups = "drop") %>%
  group_by(region, method) %>%
  mutate(rel_abundance = total_count / sum(total_count))

# Stacked bar plot
genus_stacked <- ggplot(df_rel, aes(x = method, y = rel_abundance, fill = phylum)) +
  geom_bar(stat = "identity", position = "stack") +
  facet_wrap(~region, ncol = 5) +
  labs(x = "Method", y = "Relative Abundance", fill = "Phylum") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none")
```

```{r}

#| label: composition-diversity-plots-family

# Compute relative abundance
df_rel <- combined_data_family %>%
  group_by(region, method, phylum) %>%
  summarise(total_count = sum(count), .groups = "drop") %>%
  group_by(region, method) %>%
  mutate(rel_abundance = total_count / sum(total_count))

# Stacked bar plot
family_stacked <- ggplot(df_rel, aes(x = method, y = rel_abundance, fill = phylum)) +
  geom_bar(stat = "identity", position = "stack") +
  facet_wrap(~region, ncol = 5) +
  labs(x = "Method", y = "Relative Abundance", fill = "Phylum") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "right")
```

```{r}

#| label: plot-composition-together

stacked_comp_plot <- cowplot::plot_grid(species_stacked, genus_stacked,family_stacked, labels = c("d", "e", "f"), ncol = 3, rel_widths = c(1,1,1.4))

ggsave(filename = "Figures/all_stacked_composition.png",
       plot = stacked_comp_plot, device = "png", dpi = 300, units = "mm", height = 200, width = 400)

stacked_comp_plot
```

```{r}
#| label: use-phylo-for-comp-species
#| eval: false
#| include: false

pseq_relabund <- microbiome::transform(phylo_rocky_eDNA_aglom_sp, "compositional")

phylum_rareaggregate_pseq <- microbiome::aggregate_rare(
  pseq_relabund, level = "phylum",
  detection = 0.01, prevalence = 5/100
)

phylum_bar_sp_eDNA <- 
microbiome::plot_composition(phylum_rareaggregate_pseq, 
                               otu.sort ="abundance") +
  xlab("Sample") + ylab("Relative abundance") +
  ggtitle("Phylum relative abundance bar chart") +
  #Change colours of Phylum to that of the "Paired" palette from colour brewer
  scale_fill_brewer("Phylum", palette = "Paired")
```

```{r}
#| label: use-phylo-for-comp-genus
#| eval: false
#| include: false

pseq_relabund <- microbiome::transform(phylo_rocky_eDNA_aglom_genus, "compositional")

phylum_rareaggregate_pseq <- microbiome::aggregate_rare(
  pseq_relabund, level = "phylum",
  detection = 0.01, prevalence = 5/100
)

phylum_bar <- 
  microbiome::plot_composition(phylum_rareaggregate_pseq, 
                               otu.sort ="abundance") +
  xlab("Sample") + ylab("Relative abundance") +
  ggtitle("Phylum relative abundance bar chart") +
  #Change colours of Phylum to that of the "Paired" palette from colour brewer
  scale_fill_brewer("Phylum", palette = "Paired")
```

```{r}

#| label: rel-abundance-plots

pseq_relabund <- microbiome::transform(phylo_MarClim_aglom_sp, "compositional")

phylum_rareaggregate_pseq <- microbiome::aggregate_rare(
  pseq_relabund, level = "phylum",
  detection = 0.01, prevalence = 5/100
)

phylum_bar_sp_visual <- 
  microbiome::plot_composition(phylum_rareaggregate_pseq, 
                               otu.sort ="abundance",
                               average_by = "country") +
  xlab("Sample") + ylab("Relative abundance") +
  ggtitle("Phylum relative abundance bar chart") +
  #Change colours of Phylum to that of the "Paired" palette from colour brewer
  scale_fill_brewer("Phylum", palette = "Paired")
```

#### NMDS plots

Species-level

```{r}

#| label: ord-jaccard-method-species

ord.nmds.jaccard <- ordinate(ps_merged_pa, method = "NMDS", distance =
                                   "jaccard")

stress_value_jaccard <- ord.nmds.jaccard$stress
```

```{r}

#| label: nmds-jaccard-method-species

NMDS_method_jaccard <- plot_ordination(ps_merged_pa, ord.nmds.jaccard, color="Method")+
  theme_classic()+ 
  annotate("text", x = Inf, y = Inf, label = paste("Stress =", round(stress_value_jaccard, 3)), 
           size = 4, hjust = 1.1, vjust = 1.5)+
  stat_ellipse(aes(group = Method), type = "t", level = 0.95, linetype = 1, size = 0.5)+
  labs(color = "Method") +
  theme(legend.position = "none") +
  xlim(-1,1)+
  scale_color_manual(values = c("eDNA" = "deepskyblue4", "Visual" = "darkseagreen"))+
  guides(colour = guide_legend(nrow = 2, byrow = TRUE)) 

NMDS_method_jaccard
```

Genus-level

```{r}

#| label: ord-jaccard-method-genus

ord.nmds.jaccard.genus <- ordinate(ps_merged_genus_pa, method = "NMDS", distance =
                                   "jaccard")

stress_value_jaccard_genus <- ord.nmds.jaccard.genus$stress
```

```{r}

#| label: nmds-jaccard-method-genus

NMDS_method_jaccard_genus <- plot_ordination(ps_merged_genus_pa, ord.nmds.jaccard.genus, color="Method")+
  theme_classic()+ 
  annotate("text", x = Inf, y = Inf, label = paste("Stress =", round(stress_value_jaccard_genus, 3)), 
           size = 4, hjust = 1.1, vjust = 1.5)+
  stat_ellipse(aes(group = Method), type = "t", level = 0.95, linetype = 1, size = 0.5)+
  labs(color = "Method") +
  scale_color_manual(values = c("eDNA" = "deepskyblue4", "Visual" = "darkseagreen"))+
  theme(legend.position = "none") +
  xlim(-1,1)+
  guides(colour = guide_legend(nrow = 2, byrow = TRUE)) 

NMDS_method_jaccard_genus
```

Family-level

```{r}

#| label: ord-jaccard-method-family

ord.nmds.jaccard.family <- ordinate(ps_merged_family_pa, method = "NMDS", distance =
                                   "jaccard")

stress_value_jaccard_family <- ord.nmds.jaccard.family$stress
```

```{r}

NMDS_method_jaccard_family <- plot_ordination(ps_merged_family_pa, ord.nmds.jaccard.family, color="Method")+
  theme_classic()+ 
  annotate("text", x = Inf, y = Inf, label = paste("Stress =", round(stress_value_jaccard_family, 3)), 
           size = 4, hjust = 1.1, vjust = 1.5)+
  stat_ellipse(aes(group = Method), type = "t", level = 0.95, linetype = 1, size = 0.5)+
  labs(color = "Method") +
  theme(legend.position = "right") +
  scale_color_manual(values = c("eDNA" = "deepskyblue4", "Visual" = "darkseagreen"))+
  guides(colour = guide_legend(nrow = 2, byrow = TRUE)) 

NMDS_method_jaccard_family
```

```{r}

# plot beta together

beta_all <- cowplot::plot_grid(species_stacked, genus_stacked,family_stacked,NMDS_method_jaccard, NMDS_method_jaccard_genus,NMDS_method_jaccard_family, labels = c("a", "b", "c", "d", "e", "f"), ncol = 3, rel_widths = c(1,1,1.4))

ggsave(filename = "Figures/beta_all.png",
       plot = beta_all, device = "png", dpi = 300, units = "mm", height = 300, width = 450)

beta_all
```

## Species with shifting ranges

```{r}

#| label: define-marclim-genera

changing_taxa_list <- c("Semibalanus balanoides",
                        "Alaria esculenta",
                        "Phorcus lineatus",
                        "Steromphala umbilicalis",
                        "Chthamalus montagui",
                        "Perforatus perforatus", 
                        "Patella depressa", 
                        "Bifurcaria bifurcata")

changing_taxa_list_visual <- c("semibalanus balanoides",
                        "alaria esculenta",
                        "phorcus lineatus",
                        "steromphala umbilicalis",
                        "chthamalus montagui",
                        "perforatus perforatus", 
                        "patella depressa", 
                        "bifurcaria bifurcata")
```

```{r}

#| label: marclim-taxa-regions-figures

phylo_rocky_eDNA_comp <- microbiome::transform(phylo_rocky_eDNA, "compositional")

for (i in changing_taxa_list) {
  tryCatch({
    # Subset the taxa for the current genus
    taxa <- subset_taxa(phylo_rocky_eDNA_comp, scientificname == i)
    
    # Check if there are any taxa after subsetting
    if (ntaxa(taxa) == 0) {
      stop(paste("No species found:", i))  # Manually trigger an error if no taxa are found
    }
    
    # Generate the bar plot for the current genus
    p <- plot_bar(taxa, x = "country", fill = "country") + 
      labs(y = "Read count", x = "Region") +
      theme_bw()+
      theme(legend.position = "none")
    p
    
    # Save the plot to the folder
    plot_file_path <- paste0("Figures/Changing_Species_Regions/", i, "_regions.png")
    ggsave(filename = plot_file_path, plot = p, width = 9, height = 7, dpi = 300)  # Adjust size as needed
    
  }, error = function(e) {
    # Handle the error: print a message and skip to the next genus
    message(paste("Error with genus:", i, " - ", e$message))
  })
}
```

```{r}

phylo_MarClim_comp <- microbiome::transform(phylo_MarClim, "compositional")

# Initialize empty list to store plots
plot_list <- list()

for (i in changing_taxa_list_visual) {
  tryCatch({
    # Subset the taxa for the current genus
    taxa <- subset_taxa(phylo_MarClim_comp, sciname == i)
    
    # Check if there are any taxa after subsetting
    if (ntaxa(taxa) == 0) {
      stop(paste("No species found:", i))  # Manually trigger an error if no taxa are found
    }
    
    # Generate the bar plot: counts of sciname by region, fill = region
    p <- plot_bar(taxa, x = "country", fill = "country") +
      labs(title = i, y = "Count of sciname", x = "Region") +
      theme_bw() +
      theme(legend.position = "right")
    
    # Save to the list, using the species name as the list element name
    plot_list[[i]] <- p
    
    # Optionally save individual plot files (if still needed)
    plot_file_path <- paste0("Figures/Changing_Species_Regions/", i, "_regions_Visual.png")
    ggsave(filename = plot_file_path, plot = p, width = 9, height = 7, dpi = 300)
    
  }, error = function(e) {
    message(paste("Error with genus:", i, " - ", e$message))
  })
}

# Later: Combine them using cowplot::plot_grid
# (e.g., show first 4 plots in a grid)
combined_plot <- plot_grid(plotlist = plot_list[1:4], ncol = 2)
print(combined_plot)
```

```{r}

# colours

region_colours = c(
  "Scotland" = "darkseagreen",
  "North Wales" = "lightpink3",
  "Northeast England" = "yellow3",
  "South Wales" = "lightblue",
  "Southwest England" = "mediumpurple"
)

# subset
df_rel <- combined_data %>%
  group_by(region, method, taxa) %>%
  summarise(total_count = sum(count), .groups = "drop") %>%
  group_by(region, method) %>%
  mutate(rel_abundance = total_count / sum(total_count))

df_rel <- df_rel %>%
  mutate(rel_abundance = if_else(method == "Visual", total_count, rel_abundance))

df_rel_sp1 <- df_rel %>% 
  subset(taxa == "Semibalanus balanoides")

df_rel_sp2 <- df_rel %>% 
  subset(taxa == "Phorcus lineatus")

df_rel_sp3 <- df_rel %>% 
  subset(taxa == "Steromphala umbilicalis")

df_rel_sp4 <- df_rel %>% 
  subset(taxa == "Chthamalus montagui")

df_rel_sp5 <- df_rel %>% 
  subset(taxa == "Perforatus perforatus")

df_rel_sp6 <- df_rel %>% 
  subset(taxa == "Patella depressa")

df_rel_sp7 <- df_rel %>% 
  subset(taxa == "Bifurcaria bifurcata")

# plot
sp1_plot <- ggplot(df_rel_sp1, aes(y = rel_abundance, x = region, fill = region)) +
  geom_bar(stat = "identity") +
  theme_bw() +
  scale_fill_manual(values = region_colours) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none",
        plot.title = element_text(size = 10, face = "italic")) +
  labs(fill = "Region", y = "Abundance", x = "Region", title = "Semibalanus balanoides") +
  facet_wrap(~method, ncol = 1, scales = "free_y")

sp2_plot <- ggplot(df_rel_sp2, aes(y = rel_abundance, x = region, fill = region)) +
  geom_bar(stat = "identity") +
  theme_bw() +
    scale_fill_manual(values = region_colours) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none",
        plot.title = element_text(size = 10, face = "italic")) +
  labs(fill = "Region", y = NULL, x = "Region", title = "Phorcus lineatus") +
  facet_wrap(~method, ncol = 1, scales = "free_y")

sp3_plot <- ggplot(df_rel_sp3, aes(y = rel_abundance, x = region, fill = region)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = region_colours) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none",
        plot.title = element_text(size = 10, face = "italic")) +
  labs(fill = "Region", y = NULL, x = "Region", title = "Steromphala umbilicalis") +
  facet_wrap(~method, ncol = 1, scales = "free_y")

sp4_plot <- ggplot(df_rel_sp4, aes(y = rel_abundance, x = region, fill = region)) +
  geom_bar(stat = "identity") +
  theme_bw() +
  scale_fill_manual(values = region_colours) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none",
        plot.title = element_text(size = 10, face = "italic")) +
  labs(fill = "Region", y = NULL, x = "Region", title = "Chthamalus montagui") +
  facet_wrap(~method, ncol = 1, scales = "free_y")

sp5_plot <- ggplot(df_rel_sp5, aes(y = rel_abundance, x = region, fill = region)) +
  geom_bar(stat = "identity") +
  theme_bw() +
  scale_fill_manual(values = region_colours) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none",
        plot.title = element_text(size = 10, face = "italic")) +
  labs(fill = "Region", y = NULL, x = "Region", title = "Perforatus perforatus") +
  facet_wrap(~method, ncol = 1, scales = "free_y")

sp6_plot <- ggplot(df_rel_sp6, aes(y = rel_abundance, x = region, fill = region)) +
  geom_bar(stat = "identity") +
  theme_bw() +
  scale_fill_manual(values = region_colours) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none",
        plot.title = element_text(size = 10, face = "italic")) +
  labs(fill = "Region", y = NULL, x = "Region", title = "Patella depressa") +
  facet_wrap(~method, ncol = 1, scales = "free_y")

sp7_plot <- ggplot(df_rel_sp7, aes(y = rel_abundance, x = region, fill = region)) +
  geom_bar(stat = "identity") +
  theme_bw() +
  scale_fill_manual(values = region_colours) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none",
        plot.title = element_text(size = 10, face = "italic")) +
  labs(fill = "Region", y = NULL, x = "Region", title = "Bifurcaria bifurcata") +
  facet_wrap(~method, ncol = 1, scales = "free_y")

species_all <- cowplot::plot_grid(sp1_plot, sp2_plot, sp3_plot, sp4_plot, sp5_plot, sp6_plot, sp7_plot, ncol = 4, labels = c("a", "b", "c", "d", "e", "f", "g"))

ggsave(plot = species_all, filename = "Figures/changing_species_comp_plot.png", device = "png", dpi = 300, units = "mm", height = 250, width = 250)

```

## Abundance estimates

```{r}

#calculate richness
method_count <- combined_data %>% select(count, method)

# Add an identifier column to ensure pairing
df <- method_count %>%
  group_by(method) %>%
  mutate(id = row_number()) %>%
  ungroup()

# Reshape data to wide format
df_wide <- df %>%
  pivot_wider(names_from = method, values_from = count) %>%
  select(-id) %>% # Remove the identifier column if not needed
  subset(eDNA >= 1 & Visual >= 1) %>% #remove absences
  as.data.frame()    # Convert to data frame

# Create scatter plot
ggplot(df_wide, aes(x = log(eDNA), y = Visual)) +
  geom_point(color = "darkgreen") +
  labs(
    x = "eDNA reads (logged)",
    y = "Visual counts"
  ) +
  theme_classic()
```

## Concluding remarks

We have now created all figures and reproduced all analyses presented in the manuscript.
